src/Builder/AbstractFormContractor.php

```diff
@@ -191,6 +210,26 @@ private function isAnyInstanceOf(?string $type, array $classes): bool
             }
         }
 
+        // NEXT_MAJOR: remove if condition
+        if (null !== $this->formRegistry) {
+            // handle form type inheritance and check all parent types
+            $resolvedType = $this->formRegistry->getType($type);
+            while (null !== $resolvedType->getParent()) {
+                $resolvedType = $resolvedType->getParent();
+                $parentType = \get_class($resolvedType->getInnerType());
+
+                // all types have "Symfony\Component\Form\Extension\Core\Type\FormType" as parent
+                // so we ignore it here for performance reasons
+                if (FormType::class !== $parentType) {
+                    foreach ($classes as $class) {
```

Comment:
```
return $this-> isAnyInstanceOf($parentType, $class, true)
```

this way it works if I extend a form which extends a form which extends MOdelType, isnt it ?

Reply:
Indeed using recursion makes more sense than my while loop :+1: